File: service/src/main/java/com/github/goldy1992/mp3player/dagger/modules/service/MediaSessionCompatModule.java
package com.github.goldy1992.mp3player.dagger.modules.service;

import android.content.Context;
import android.support.v4.media.session.MediaSessionCompat;

import com.github.goldy1992.mp3player.service.player.MyMediaButtonEventHandler;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class MediaSessionCompatModule {

    private final String LOG_TAG = "MEDIA_SESSION_COMPAT";

    @Singleton
    @Provides
    public MediaSessionCompat provideMediaSessionCompat(Context context) {
        return new MediaSessionCompat(context, LOG_TAG);
    }

    @Singleton
    @Provides
    public MediaSessionCompat.Token provideMediaSessionToken(MediaSessionCompat mediaSessionCompat) {
        return mediaSessionCompat.getSessionToken();
    }

    @Provides
    @Singleton
    public MyMediaButtonEventHandler myMediaButtonEventHandler() {
        return new MyMediaButtonEventHandler();
    }
}

File: service/src/main/java/com/github/goldy1992/mp3player/service/dagger/modules/service/MediaSessionCompatModule.kt
package com.github.goldy1992.mp3player.service.dagger.modules.service

import android.content.Context
import android.support.v4.media.session.MediaSessionCompat
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ServiceComponent
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.android.scopes.ServiceScoped

@InstallIn(ServiceComponent::class)
@Module
class MediaSessionCompatModule {
    private val LOG_TAG = "MEDIA_SESSION_COMPAT"

    @Provides
    @ServiceScoped
    fun provideMediaSessionCompat(@ApplicationContext context: Context): MediaSessionCompat {
        return MediaSessionCompat(context, LOG_TAG)
    }

    @ServiceScoped
    @Provides
    fun provideMediaSessionToken(mediaSessionCompat: MediaSessionCompat): MediaSessionCompat.Token {
        return mediaSessionCompat.sessionToken
    }

}